### PR TITLE
creating a data model

### PR DESCRIPTION
**Pull request by banzaman**
### Project requirements
Be sure to reference the [ERD diagram](../images/blog_app_erd.png) from the sneak peek if you have any questions about table or column names or data relationships.

- Create a new app. Give it a meaningful name.
- Initialize your project with Git.
- Make sure that your project has Postgres database set up.
- Build your project schema.
    - Create and run the necessary migration files.
    - Table and column names should match the ERD diagram.
        - note: photo for users table should be a link to an image
    - Foreign keys should be included.
        - Pay attention that in `posts` table the column holding the foreign key to the `users` table should be named `author_id`
    - All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

File: db/migrate/20230607203415_create_likes.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes, &:timestamps
+  end
+end
```
Comment (AmaduKamara) [inline db/migrate/20230607203415_create_likes.rb]: - [ ] Kindly set up references for the author and post.

```ruby
create_table :likes do |t|
    t.timestamps 
    t.references :author, foreign_key: { to_table: 'users' } 
    t.references :post, foreign_key: true 
  end
```

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @banzaman,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [x] According to the project requirement, you're required to use a PostgreSQL adaptor. In your database.yml file, you're using SQLite. Kindly use PostgreSQL and the adapters should look like this: 
```ruby
default: &default
adapter: postgresql
encoding: unicode
```

Screenshot 2023-06-07 at 9 28 19 PM

- [x] You could achieve that by creating a new app with the command `rails new myapp --database=postgresql`

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: db/migrate/20230607203415_create_likes.rb (hunk 1 above)
Comment (banzaman) [reply db/migrate/20230607203415_create_likes.rb]: How did i forget that 
let me do that

Comment (banzaman) [reply db/migrate/20230607203415_create_likes.rb]: let me work on that

**Review by mwanawabangona — APPROVED** [review]
Hi @banzaman,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by banzaman** [comment]
thanks for your time @AmaduKamara & @mwanawabangona

**Review by topeogunleye — COMMENTED** [review]
Hi @banzaman, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

dwayne-the-rock

## (Highlights) Good Points: 💪 
- The student has successfully created the **users**, **posts**, **comments**, and **likes** tables using **Postgresql** as the database adapter. 👏
- The student has used appropriate data types and constraints for each column, such as **string**, **text**, **integer**, **timestamps**, and **foreign_key**. 👍
- The student has followed the **ERD diagram** and established the correct relationships between the tables using **references**. 🙌
- The student has used clear and consistent naming conventions for the tables and columns, such as **author_id** for the foreign key in the **posts** table. 🤩

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

You can also consider:
- Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**